miri-script/src/main.rs

```diff
@@ -150,108 +209,15 @@ CARGO_EXTRA_FLAGS:
-                } else if let Some(val) = args.get_long_opt_with_default("many-seeds", "0..64")? {
-                    let (from, to) = val.split_once("..").ok_or_else(|| {
-                        anyhow!("invalid format for `--many-seeds`: expected `from..to`")
-                    })?;
-                    let from: u32 = if from.is_empty() {
-                        0
-                    } else {
-                        from.parse().context("invalid `from` in `--many-seeds=from..to")?
-                    };
-                    let to: u32 = to.parse().context("invalid `to` in `--many-seeds=from..to")?;
-                    many_seeds = Some(from..to);
-                } else if let Some(val) = args.get_long_opt("target")? {
-                    target = Some(val);
-                } else if let Some(val) = args.get_long_opt("edition")? {
-                    edition = Some(val);
-                } else if let Some(flag) = args.get_other() {
-                    flags.push(flag);
-                } else {
-                    break;
-                }
-            }
-            Command::Run { dep, verbose, many_seeds, target, edition, flags }
-        }
-        Some("fmt") => Command::Fmt { flags: args.remainder() },
-        Some("clippy") => Command::Clippy { flags: args.remainder() },
-        Some("install") => Command::Install { flags: args.remainder() },
-        Some("bench") => {
-            let mut target = None;
-            let mut benches = Vec::new();
-            loop {
-                if let Some(val) = args.get_long_opt("target")? {
-                    target = Some(val);
-                } else if let Some(flag) = args.get_other() {
-                    benches.push(flag);
-                } else {
-                    break;
-                }
-            }
-            Command::Bench { target, benches }
-        }
-        Some("toolchain") => Command::Toolchain { flags: args.remainder() },
-        Some("rustc-pull") => {
-            let commit = args.next_raw();
-            if args.next_raw().is_some() {
-                bail!("Too many arguments for `./miri rustc-pull`");
-            }
-            Command::RustcPull { commit }
-        }
-        Some("rustc-push") => {
-            let github_user = args.next_raw().ok_or_else(|| {
-                anyhow!("Missing first argument for `./miri rustc-push GITHUB_USER [BRANCH]`")
-            })?;
-            let branch = args.next_raw().unwrap_or_else(|| "miri-sync".into());
-            if args.next_raw().is_some() {
-                bail!("Too many arguments for `./miri rustc-push GITHUB_USER BRANCH`");
-            }
-            Command::RustcPush { github_user, branch }
-        }
-        _ => {
-            eprintln!("Unknown or missing command. Usage:\n\n{HELP}");
-            std::process::exit(1);
-        }
-    };
+    let miri_args: Vec<_> =
+        std::env::args().peekable().take_while(|x| *x != "--").filter(|x| *x != "--").collect();
```

Comment:
Why `peekable`? And why the extra `filter`?

Reply:
Yeap, `peekable` is not needed. I just quickly copy pasted the code from the `args` module without thinking too much for the prototype. I used the extra  filter because for some reason i was under the impression that the `--` was included in the result. This is wrong and the filter is not needed.